### PR TITLE
DB: Fix time handling

### PR DESCRIPTION
**Pull request by ddutt**
The DB is optimized to provide constant query times independent of the
time window specified. The code to read the set of files given the
variety of user inputs was buggy and read a lot more files than
necessary. This reduced the time to respond to user queries. This patch
addresses this.

Signed-off-by: Dinesh Dutt <dd.ps4u@gmail.com>

### CODE REVIEW

File: suzieq/db/parquet/parquetdb.py [hunk 1]
```diff
@@ -509,16 +509,8 @@ def _get_cp_dataset(self, table_name: str, need_sqvers: bool,
...
-            # Enforcing the logic we have: if both start_time & end_time
-            # are given, return all files since the model is that the user is
-            # expecting to see all changes in the time window. Otherwise, user
-            # expecting to see only the latest before an end_time OR after a
-            # start_time.
-            all_files = True
-        else:
-            all_files = False
+        # pylint: disable=consider-using-ternary
+        all_files = ((start_time and end_time) or (view == "all"))
```
Comment (LucaNicosia) [inline suzieq/db/parquet/parquetdb.py:513]: all_files is true only for both start_time and end_time are set (or view == 'all'). Shouldn't we set all_files to False also if only one between start_time or end_time are specified?
Isn't it `all_files = (not (start_time or end_time) or (view == 'all'))` what we should do?

Comment (claudious96) [reply suzieq/db/parquet/parquetdb.py:513]: I agree with Luca, with this change if we set both `start_time` and `end_time` we are going to read every file of the dataset.

Comment (ddutt) [reply suzieq/db/parquet/parquetdb.py:513]: I agree, Ugh. I had the original code, but it'd read all files if we only gave end-time. I don't the logic is here that needs to be fixed.

Comment (ddutt) [reply suzieq/db/parquet/parquetdb.py:513]: The correct logic I think is this in English; if view = all is specified, and there are no time parameters, just suck in all files. So, its (view == "all") and not (start_time or end_time)

**Review by claudious96 — APPROVED** [review]

**Comment by claudiolor** [comment]
I know the PR is already merged, but I left the comments anyway, if needed we can open another PR to address these problems. I left a comment about view=all and only the end-time filter. Additionally, some comments about additional comments and variable rename, as I add some difficulties with understanding the code while doing the review

**Pull request merged**